src/main/java/by/yurachel/web_app/repository/PhoneRepository.java

```diff
@@ -46,8 +48,47 @@ public void changePhoneParam(Phone newPhone, String oldPhoneName) {
         }
     }
 
-    public void removePhone(Phone phone) {
-        getPhones().remove(phone);
+    public void removePhone(int id) {
+       boolean isRemove =  phones.removeIf(phone -> phone.getId() == id);
+       if(!isRemove){
+           throw new IllegalArgumentException("The phone wasn't found");
+       }
 
     }
+
+    public Phone findPhoneById(int id) {
+        Phone findPhone = new Phone();
+        for (Phone phone : phones) {
+            if (phone.getId() == id) {
+                findPhone = phone;
+                break;
+            }
+        }
+        return findPhone;
+    }
+
+
+    public int findPhoneIDByName(String name) {
+        Integer findIdPhone = null;
+        for (Phone phone : phones) {
+            if (phone.getName().equals(name)) {
+                findIdPhone = phone.getId();
+                break;
+            }
+        }
+        if (findIdPhone == null) {
+            throw new NullPointerException("Such phone wasn't found");
```

Comment:
**Never throw NPE manually**, there should be any other exception (better approach is custom exception), or at least IllegalArgumentException